inc/class-wpseo-admin-bar-menu.php

```diff
@@ -457,7 +457,7 @@ protected function get_singular_post() {
 			return null;
 		}
 
-		if ( ! isset( $post ) || ! is_object( $post ) ) {
+		if ( ! isset( $post ) || ! is_object( $post ) || ! is_a( $post, 'WP_Post' )  ) {
```

Comment:
The `is_a` function is nice, however because of the PHP 5.2 support I prefer using an `instanceof` check here. `is_a` was deprecated in PHP 5.0.0 and it came back in 5.3.0. 
```suggestion
		if ( ! isset( $post ) || ! is_object( $post ) || ! $post instanceof WP_Post )  ) {
```